commit 8899704ac983f248deb84c40d3570b28aac4e94e (refs/changes/05/2605/2)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2017-09-19T14:03:53-06:00 (2 years, 1 month ago)
    
    joyent/node-manta#328 Version and Help should not require Vars to be Set

diff --git a/lib/options.js b/lib/options.js
index 3fbffed..4aa6caa 100644
--- a/lib/options.js
+++ b/lib/options.js
@@ -48,9 +48,8 @@ function parseOptions(args) {
 
     try {
         opts = parser.parse(process.argv);
-        cc.checkBinEnv(opts);
     } catch (e) {
-        cc.usage(parser, e.message, 'path...');
+            cc.usage(parser, e.message, 'path...');
     }
 
     cc.setupLogger(opts, args.log);
@@ -78,5 +77,11 @@ function parseOptions(args) {
 
     args.parseCmdOptions(opts, parser);
 
+    try {
+        cc.checkBinEnv(opts);
+    } catch (e) {
+        cc.usage(parser, e.message, 'path...');
+    }
+
     return (opts);
 }
\ No newline at end of file
diff --git a/test/mput.test.js b/test/mput.test.js
index 4a32f53..0c16bda 100644
--- a/test/mput.test.js
+++ b/test/mput.test.js
@@ -169,6 +169,48 @@ test('mput with custom header value with colons', function (t) {
 });
 
 
+/*
+ * Test that specifying the --help option with no manta URL specified does not
+ * result in a warning about the missing URL. This verifies the fix for
+ * https://github.com/joyent/node-manta/issues/328.
+ */
+test('mput --help with no manta URL specified', function (t) {
+    forkExecWait({
+        argv: [ MPUT,
+                '--help'
+              ],
+        env: { MANTA_URL: '' }
+    }, function (err, info) {
+           t.ifError(err, err);
+
+           const urlArgMsg = 'url is a required argument';
+           t.equal(info.stderr.indexOf(urlArgMsg), -1);
+           t.done();
+    });
+});
+
+
+/*
+ * Test that specifying the --version option with no manta URL specified does
+ * not result in a warning about the missing URL. This verifies the fix for
+ * https://github.com/joyent/node-manta/issues/328.
+ */
+test('mput --version with no manta URL specified', function (t) {
+    forkExecWait({
+        argv: [ MPUT,
+                '--version'
+              ],
+        env: { MANTA_URL: '' }
+    }, function (err, info) {
+           t.ifError(err, err);
+
+           const urlArgMsg = 'url is a required argument';
+           t.equal(info.stderr.indexOf(urlArgMsg), -1);
+           t.done();
+    });
+});
+
+
 test('cleanup: rm test tree ' + TESTDIR, function (t) {
     // Sanity checks that we don't `mrm -r` a non-test dir.
     assert.ok(TESTDIR);
